pkg/repositories/gormimpl/common.go

```diff
@@ -60,6 +62,32 @@ var leftJoinTaskToTaskExec = fmt.Sprintf(
 	taskTableName, taskExecutionTableName, taskTableName, taskExecutionTableName, taskTableName,
 	taskExecutionTableName, taskTableName, taskExecutionTableName, taskTableName)
 
+// Creates a JOIN clause between a table which can hold NamedEntityIdentifiers
+// and the metadata associated between them. Metadata is optional, so a left
+// join is used.
+func CreateEntityMetadataJoin(entityTableName string, resourceType core.ResourceType) string {
```

Comment:
Please add tests for all these new methods too!

Reply:
I'm not certain a test is necessary for this one. It's only used to generate three string constants that are used in queries which are already covered under testing in the NamedEntityRepo class. I actually might just remove the helper function and make the three constants stand on their own

Reply:
But I will add some for the other methods!